src/Avalonia.Controls/RepeatButton.cs

```diff
@@ -23,20 +29,39 @@ public int Delay
             set { SetValue(DelayProperty, value); }
         }
 
+        /// <summary>
+        /// Gets or sets the amount of time, in milliseconds, to wait before repeating begins.
+        /// </summary>
+        public int InitialDelay
+        {
+            get { return GetValue(InitialDelayProperty); }
+            set { SetValue(InitialDelayProperty, value); }
+        }
+
         private void StartTimer()
         {
             if (_repeatTimer == null)
             {
                 _repeatTimer = new DispatcherTimer();
-                _repeatTimer.Tick += (o, e) => OnClick();
+                _repeatTimer.Tick += RepeatTimerOnTick;
             }
 
             if (_repeatTimer.IsEnabled) return;
 
-            _repeatTimer.Interval = TimeSpan.FromMilliseconds(Delay);
+            _repeatTimer.Interval = TimeSpan.FromMilliseconds(InitialDelay);
             _repeatTimer.Start();
         }
 
+        private void RepeatTimerOnTick(object sender, EventArgs e)
+        {
+            var interval = TimeSpan.FromMilliseconds(Delay);
+            if (_repeatTimer.Interval != interval)
```

Comment:
We should probably modify `DispatcherTimer` so that setting the `Interval` to the same value is a no-op.